### PR TITLE
Adding 1.23 kind test

### PR DESCRIPTION
**Pull request by yuriolisa**
@jpkrohling

### CODE REVIEW

**Comment by jpkrohling** [comment]
There's already a PR for this, #617 :-)

File: README.md [hunk 1]
```diff
@@ -229,9 +229,9 @@ The OpenTelemetry Operator *might* work on versions outside of the given range,
 
 | OpenTelemetry Operator | Kubernetes           | Cert-Manager         |
 |------------------------|----------------------|----------------------|
-| v0.40.0                | v1.20 to v1.22       | 1.6.1                |
-| v0.39.0                | v1.20 to v1.22       | 1.6.1                |
-| v0.38.0                | v1.20 to v1.22       | 1.6.1                |
+| v0.40.0                | v1.20 to v1.23       | 1.6.1                |
+| v0.39.0                | v1.20 to v1.23       | 1.6.1                |
```
Comment (jpkrohling) [inline README.md:233]: That's not accurate: we didn't test 0.39.0 with 1.23. We will start with 0.41.0.

Comment (yuriolisa) [reply README.md:233]: Ok, I tested from `0.37.0` to `0.40.0` and only from `0.38.0` worked well.

Comment (jpkrohling) [reply README.md:233]: Right, but I mean, the compat matrix should really only change for 0.41.0, dropping one version and adding the new one.

**Comment by VineethReddy02** [comment]
There is already PR #615 on the same. So closing it in reference to that PR. :)

**Pull request closed without merging**